editable.c

```diff
@@ -247,8 +247,8 @@ static void move_sel(struct editable *e, struct list_head *after)
 	editable_track_to_iter(e, to_simple_track(after->next), &iter);
 
 	if (editable_owns_shared(e)) {
-		window_set_sel(e->shared->win, &iter);
```

Comment:
Could you explain what exactly causes the crash when window_set_sel is called first?

Reply:
Immediate cause of the crash is line 350 in window.c, in function `window_set_sel` (used to set the selected row):

    BUG_ON(iter_is_empty(&win->top));

This asserts that the window must not be showing an empty list  (`win->top` points to the item in the window’s topmost row, and is empty if there are no items in the list). This variable was not updated correctly when all tracks in a list were moved, so the window appeared to contain an empty list when trying to set the selected row.

Moving tracks to another position in a list works by first removing all marked items and then reinserting them at the new position. For each removed item, `window_row_vanishes` is (correctly) called to update `win->top`. After adding all the items back to the list, `window_changed` should be called for the same reason.

Existing code, however, called `window_changed` only after `window_set_sel`. This means that `win->top` was empty (if all tracks were moved) during the call to `window_set_sel`, and the above assertion failed.

Calling `window_changed` first ensures that `top` points to an actual (first) row, and the check for non-empty list succeeds when setting selection. Compare this with functions `editable_sort` and `editable_rand`, where `window_changed` is also called before setting the selected row.